Bottom: 78bf4fbf65192d6495b3d0d171e1a40fc82b93a4
Top:    7125f7f07015d1cbe41b6ec0158eb62b0a81507d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 01:22:14 +0100

Refresh of try-something-with-the-rtrees

---

diff --git a/src/polygon1.c b/src/polygon1.c
index df7e2ee..1faf196 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1626,7 +1626,11 @@ M_B_AREA_Collect (jmp_buf * e, POLYAREA * bfst, POLYAREA ** contours,
 		next = cur;
 		tmp->next = NULL;
 		tmp->Flags.status = UNKNWN;
-		PutContour (e, tmp, contours, holes, NULL, NULL, NULL); /* b */
+		/* Strictly we should pass b as the "owner" parameter, but
+		 * since the b POLYAREA is going to be delete anyway, it
+		 * doesn't matter that its contour r-tree gets out of sync.
+		 */
+		PutContour (e, tmp, contours, holes, NULL /* b */, NULL, NULL);
 		break;
 	      case PBO_UNITE:
 		break;		/* nothing to do - already included */
@@ -1642,7 +1646,11 @@ M_B_AREA_Collect (jmp_buf * e, POLYAREA * bfst, POLYAREA ** contours,
 		next = cur;
 		tmp->next = NULL;
 		tmp->Flags.status = UNKNWN;
-		PutContour (e, tmp, contours, holes, NULL, NULL, NULL); /* b */
+		/* Strictly we should pass b as the "owner" parameter, but
+		 * since the b POLYAREA is going to be delete anyway, it
+		 * doesn't matter that its contour r-tree gets out of sync.
+		 */
+		PutContour (e, tmp, contours, holes, NULL /* b */, NULL, NULL);
 		break;
 	      case PBO_ISECT:
 	      case PBO_SUB:
